tests/plans/test_scanspec.py

```diff
@@ -0,0 +1,98 @@
+from pathlib import Path
+from typing import cast
+from unittest.mock import patch
+
+import pytest
+from bluesky.run_engine import RunEngine
+from event_model.documents import (
+    DocumentType,
+    Event,
+    EventDescriptor,
+    RunStart,
+    RunStop,
+)
+from ophyd_async.core import (
+    DeviceCollector,
+    PathProvider,
+    StandardDetector,
+)
+from ophyd_async.sim.demo import PatternDetector, SimMotor
+from scanspec.specs import Line
+
+from dodal.plans import spec_scan
+
+
+@pytest.fixture
+def det(RE: RunEngine, tmp_path: Path) -> StandardDetector:
+    with DeviceCollector(mock=True):
+        det = PatternDetector(tmp_path / "foo.h5")
+    return det
+
+
+@pytest.fixture
+def x_axis(RE: RunEngine) -> SimMotor:
+    with DeviceCollector(mock=True):
+        x_axis = SimMotor()
+    return x_axis
+
+
+@pytest.fixture
+def y_axis(RE: RunEngine) -> SimMotor:
+    with DeviceCollector(mock=True):
+        y_axis = SimMotor()
+    return y_axis
+
+
+@pytest.fixture
+def path_provider(static_path_provider: PathProvider):
+    # Prevents issue with leftover state from beamline tests
+    with patch("dodal.plan_stubs.data_session.get_path_provider") as mock:
+        mock.return_value = static_path_provider
+        yield
+
+
+def test_output_of_simple_spec(
```

Comment:
Nit: I'm not a big fan of big tests like this that are testing loads of things at once as I think they are poor documentation and make it harder to know what exactly the problem is when they break. I would prefer smaller tests like `test_scan_spec_gives_run_start_with_expected_shape` etc.

Reply:
My $0.02: It's fine as long as you do both, a big integration test is sometimes good too as it catches issues that only arise when you put everything together, however a small unit test is _more_ useful for the reasons you say so if you can only do one, do that.

On the other hand I think we do have to keep a close eye on how long it takes to run the dodal test suite. As we add more and more beamlines I'm expecting it to increase significantly and we might need to start thinking about solutions like [pytest-parallel](https://pypi.org/project/pytest-parallel/). I'd like to avoid that as long as possible though because we might get lazy and complacent if we know we can always get more resources (see every web browser).

Reply:
I agree that both is best if possible, we have some tests in `mx-bluesky` that are larger integration ones alongside the ones just testing small bits of logic. I also agree with the idea that they shouldn't take too long and I do regularly add PRs that just speed them up a bit. However, in my experience long running tests are mostly a consequence of poor mocking rather than quantity and it's actually large integration tests that end up taking longer than many small ones as people tend to have done more mocking in the smaller tests. Tests currently take ~30s on `dodal`, which seems reasonable. I would suggest if they start hitting ~60s we should think harder about this.